processing/src/main/java/org/apache/druid/segment/join/filter/JoinFilterPreAnalysis.java

```diff
@@ -51,8 +52,7 @@
   private final List<Filter> normalizedBaseTableClauses;
   private final List<Filter> normalizedJoinTableClauses;
   private final JoinFilterCorrelations correlations;
-  private final boolean enableFilterPushDown;
-  private final boolean enableFilterRewrite;
+  private final JoinFilterPreAnalysisGroup myGroup;
```

Comment:
Does this introduce a memory leak / a circular dependency of some kind?

The group holds a reference to the pre-analysis object via the `analyses` concurrent map and the pre-analyses object holds a reference back to the group object via `myGroup`

Reply:
I think it's fine, but I restructured this to store all the config parameters in a new `JoinFilterRewriteConfig` class, so there's no more reference to the group in the pre-analysis objects.